### DISCUSSION TITLE
Formatting namespace imports

### DISCUSSION THREAD

**Discussion by nixin72** [kkinnear/zprint#286] [Q&A]
Is there a way to format namespace imports such that each class is on it's own line? For example 

```
(ns ex
  (:import 
   [java.io File]
   [java.util Date Arrays]))
```
Would become 
```
(ns ex
  (:import 
   [java.io 
    File]
   [java.util 
    Date 
    Arrays]))
```

**Comment by kkinnear** [comment]
Thanks for asking.  

I can certainly make your example work.  I'm not sure how well this will work for a more complex example, in that I don't know what you would want that more complex example to look like, but see if this will do what you want:
```
(czprint i286 {:parse-string? true :fn-map {":import" [:flow {:vector {:force-nl? true}}]}})
(ns ex
  (:import
    [java.io
     File]
    [java.util
     Date
     Arrays]))
```
You would want to change the `:fn-map` and add the value for `":import"` as I have in the example above.  Let me know if this doesn't do what you want -- and send a more complex example of what you do want it to look like, and I'll see what I can do.

**Reply by nixin72** [reply]
Thank you for your reply! I ended up finding a solution. I'm using an older fork of zprint, so `force-nl?` wasn't an option, but I ended up using 
```
  ":import" [:flow
             {:vector    {:fn-format :flow}
              :vector-fn {:indent 1}}]
```

**Comment by kkinnear** [comment]
Go you!  That's very impressive.  Few people have figured out the `{:vector {:fn-format ...}}` thing, despite it being documented.  I have assumed that mostly I am the only person to use it when answering people's questions.  You are clearly a power user of zprint!  Thanks for telling me what you got to work.  Feel free to ask for help anytime.
